crypto/dh/dh_pmeth.c

```diff
@@ -130,25 +141,20 @@ static int pkey_dh_copy(EVP_PKEY_CTX *dst, EVP_PKEY_CTX *src)
 
     dctx->kdf_type = sctx->kdf_type;
     dctx->kdf_oid = OBJ_dup(sctx->kdf_oid);
-    if (!dctx->kdf_oid)
-        return 0;
+    if (dctx->kdf_oid == NULL)
+        goto err;
     dctx->kdf_md = sctx->kdf_md;
-    if (dctx->kdf_ukm) {
+    if (dctx->kdf_ukm != NULL) {
```

Comment:
This is a bug in the original code. The check here should be:
`if (sctx->kdf_ukm != NULL) {`